client-common/src/tendermint/rpc_client/sync_rpc_client.rs

```diff
@@ -309,4 +322,31 @@ impl Client for SyncRpcClient {
             })
             .collect()
     }
+
+    fn get_sender(&self) -> Result<WebSocketSender> {
+        Ok(self.async_rpc_client.websocket_writer.clone())
+    }
+}
+
+/// close websocket connection by sending CLOSE message
+pub fn close_connection(writer: Arc<Mutex<WebSocketWriter>>) -> Result<()> {
+    // close
+
+    let runtime = Runtime::new().chain(|| {
+        (
+            ErrorKind::InitializationError,
+            "Unable to start tokio runtime",
+        )
+    })?;
```

Comment:
Why are you creating a new `Runtime` just to close the connection?

Reply:
because `close` is actually `sending close message`, it's async , future.
so to run it, needed runtime.
sharing runtime with original is a little complex, so chose making a new tokio runtime to run the future

Reply:
If you do it in `Drop` implementation of `SyncRpcClient`, you'll also have access to the main `Runtime`?

Reply:
ok, i'll try to do it on Drop